packages/shell-api/src/database.ts

```diff
@@ -144,6 +144,17 @@ export default class Database extends ShellApiWithMongoClass {
     return this._cachedCollectionNames;
   }
 
+  async _getCollectionNamesWithTypes(options?: ListCollectionsOptions): Promise<Document[]> {
+    const collections = await this._listCollections({}, { ...options, nameOnly: true });
+    const typesToBages: any = {
+      collection: '',
+      timeseries: '[time-series]',
+      view: '[view]'
+    };
+
+    return collections.map((collection: any) => ({ name: collection.name, badge: typesToBages[collection.type] }));
```

Comment:
This should cache the collection names just like `_getCollectionNames` does, I would say, so that whenever we list collections, the cache contains the most up-to-date data.

We probably also need to handle the case that `collection.type` is something we’re not expecting yet (e.g. because it’s only added in a future version of the server), so that `badge` is actually always a string and not potentially `undefined`.

Reply:
Hm, we use `_cachedCollectionNames` for autocomplete, so I can see the cache value here. But we use `_getCollectionNamesWithTypes` only for the showing collections command, where I would expect always up-to-date values. Which use case do you have in mind for this cache?

Reply:
Right, I’m not saying that the result of `_getCollectionNamesWithTypes` should be cached, only that the `_cachedCollectionNames` list should be updated here as well.

Reply:
i.e.:
```suggestion
    this._cachedCollectionNames = collection.map((collection: any) => collection.name);
    return collections.map((collection: any) => ({ name: collection.name, badge: typesToBages[collection.type] }));
```

Reply:
Ah, got it!